weaver/wps_restapi/processes/processes.py

```diff
@@ -70,7 +72,10 @@ def execute_process(self, job_id, url, headers=None, notification_email=None):
     load_pywps_cfg(settings)
 
     ssl_verify = asbool(settings.get("weaver.ssl_verify", True))
-    store = get_db(app).get_store(StoreJobs)
+
+    # reset the connection because we are in a forked celery process
+    db = MongoDatabase(get_registry(app), reset_connection=True)
+    store = db.get_store(StoreJobs)
```

Comment:
revert to `get_db`

Reply:
I think in this context we need to reset the mongodb connection because the celery worker is forked, and the connection can't be shared. I was getting connection errors when I didn't do it. Do you have any idea why I would get connection errors?

Reply:
see: http://api.mongodb.com/python/current/faq.html#pymongo-fork-safe

Reply:
Maybe cause by the various connections between celery/weaver when entering this module?
I have no issue with the `reset_connection=True`, but I would prefer to have it as an input of `get_db`, and this one could have a `*args, **kwargs` in it signature, that it passed down to `MongoDatabase`.
I want to have all calls to db connections regrouped in this function. Otherwise we end up having patches like this left and right.

Reply:
see 8ac83b0

Reply:
need the Optional[] typing

Reply:
doesn't Optional mean it can be None?

Reply:
see: https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
oh, the more you know... will need to revert some typing in many places!